### DISCUSSION TITLE
Update dtb file on TX2 NX

### DISCUSSION THREAD

**Discussion by dat-tran-matternet** [OE4T/meta-tegra#645] [Q&A]
Hi, I'm working with the new TX2 NX module. I assume it's similar to the TX2 modules. During development with device tree modification, I would like to just change the DTB of my system. I noticed that cboot is the one that's loading the DTB as seen from these log output:
```
[0001.382] I> Loading kernel-dtb from partition
[0001.386] I> Loading partition kernel-dtb at 0x80000000 from device(0x1)
[0001.404] I> Kernel_dtb @0x80000000
```

I assume if I can find which partition it is pointing to I can overwrite the DTB file with an updated one?

thanks

**Comment by dwalkes** [comment]
I've never tried it myself but the suggestions from https://github.com/OE4T/meta-tegra/issues/401 might be helpful.

**Comment by ichergui** [comment]
Hey @dat-tran-matternet 
For your information, the `DTB` is signed then authenticated and loaded by cboot.
You can check the partition layout as well.

**Reply by dat-tran-matternet** [reply]
> Hey @dat-tran-matternet
> For your information, the `DTB` is signed then authenticated and loaded by cboot.
> You can check the partition layout as well.

In that case, does the doflash.sh support flashing just the DTB file? It would require that I be local with USB connected but I can live with that for now. 

thanks

**Reply by kekiefer** [reply]
It's kinda tricky now. I'm pretty sure you used to be able to use `signwrite` and it would handle adding the signing header for you, but this doesn't seem to work now. The following, will work, provided you replace the dtb file that would otherwise be flashed. You need to dig into the `multi_signed` directory and find the right dtb for your board rev

```
./generate_bup_payload.sh
./doflash.sh -c "write kernel-dtb multi_signed/path/blah-blah.dtb.encrypt; reboot"
```

The other option if you've run `generate_bup_payload.sh` is to copy the `payloads_19x/kernel_only_payload` over to the target, and use `nv_update_engine` or `tegra-bootloader-update` to install it.

I wish there were an easier way. If you have a chance to poke around and could update these instructions it would be very useful.

**Reply by dat-tran-matternet** [reply]
Awesome thanks!

**Reply by dat-tran-matternet** [reply]
Looking at the nvidia forum, I noticed people mention about the script "flash.sh" a lot to update dtb and cfg files. This script is not present in the tegraflash output from yocto. Is this script only available/usable if not using yocto? Just want to understand how yocto deviates from the official Jetson SDK/BSP. thanks

**Reply by dwalkes** [reply]
> Is this script only available/usable if not using yocto?

Yes, it assumes you are using NVIDIAs L4T Ubuntu based distribution.  It's replaced by the `doflash.sh` script and the instructions in https://github.com/OE4T/meta-tegra/wiki/Flashing-the-Jetson-Dev-Kit for OE4T/meta-tegra.

**Reply by ichergui** [reply]
Hey @dat-tran-matternet 
Did you manage to update the DTB on the TX2 NX ?

**Reply by dat-tran-matternet** [reply]
I haven't tried the methods discussed here yet. I have been doing it the long way by flashing a new tegraflash image every time. Luckily I haven't needed to flash the dtb a lot yet. I'll update once I try it out. thanks
